docs/integrations/spark.md

```diff
@@ -81,6 +179,180 @@ df.write.partitionBy("example-column").parquet(s"s3a://${repo}/${branch}/output-
 
 The data is now created in lakeFS as new changes in your branch. You can now commit these changes, or revert them.
 
+
+## Access lakeFS using the lakeFS-specific Hadoop FileSystem
+
+To use this mode you configure the Spark application to perform metadata operations on the
+lakeFS server, and all data operations directly through the same underlying object store that
+lakeFS uses.  The lakeFS FileSystem currently supports using only the S3A Hadoop FileSystem
+for data access.  In this mode the Spark application will directly read and write from the
+underlying object store, significantly increasing application scalability and performance by
+reducing the load on the lakeFS server.
+
+Accessing data in lakeFS from Spark is the same as accessing S3 data from Spark.  The only
+changes we need to perform are:
+1. Configure to access lakeFS for metadata and S3 or a compatible underlying object store to
+   access data.
+1. Use `lakefs://repo/ref/path/to/data` URIs to read and write data on lakeFS, rather than
+   `s3a://...` URIs.
+
+### Configuration
+
+In order to configure Spark to work using the lakeFS Hadoop FileSystems, you will need to load
+the filesystem JARs and then configure both that FileSystem and the underlying data access
+FileSystem.
+
+#### Load the FileSystem JARs
+
+Add the package `io.lakefs:lakefsfs:<VERSION>` to your Spark job.  For instance, add
+`--packages io.lakefs:lakefsfs:0.45.56` (**TODO(nopcoder):** insert some correct coordinates, also in the CLI examples below!).
+
+#### Configure the lakeFS FileSystem and the underlying S3A FileSystem
+
+Add Hadoop configuration to the underlying storage and additionally to lakeFS credentials.
+When using this mode, do **not** set the S3A endpoint URL to point at lakeFS -- it should
+point at the underlying storage.
+
+| Hadoop Configuration   | Value                                 |
+|------------------------|---------------------------------------|
+| `fs.s3a.access.key`    | Set to the AWS S3 access key          |
+| `fs.s3a.secret.key`    | Set to the AWS S3 secret key          |
+| `fs.s3a.endpoint`      | Set to the AWS S3-compatible endpoint |
```

Comment:
```suggestion
| `fs.s3a.endpoint`      | Set to the AWS S3 endpoint |
```

Reply:
@johnnyaug asked for this, and I think he's right: it works with any S3-compatible endpoint.

Reply:
I think that he asked it for the s3 gateway configurations not? 
When I read "AWS S3-compatible" I think of the lakefs endpoint which is compatible with s3. IMHO we should use "Set to the AWS S3-compatible endpoint" in the s3 gateway part, and here use "Set to the AWS S3 endpoint". 
WDYT?

Reply:
Not really.  E.g. this would work with Minio.